### ISSUE TITLE
[12.0] partner_default_sale_discount, error replacing context in field_view_get

### ISSUE THREAD

**Issue by GSLabIt** [OCA/partner-contact#1060]
```
Error:
Error: , undefined as prefix

http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:105
Traceback:
nud@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:105:80
expression@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:141:146
@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:130:9
expression@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:141:146
py.parse@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:259:462
py.eval@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:274:296
eval_contexts/<@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:410:110
iterator@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:14:183
createReduce/<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:17:8
_.mixin/</_.prototype[name]@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:69:526
eval_contexts@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:408:136
eval_contexts/<@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:410:253
iterator@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:14:183
createReduce/<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:17:8
_.mixin/</_.prototype[name]@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:69:526
eval_contexts@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:408:136
eval_contexts/<@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:410:253
iterator@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:14:183
createReduce/<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:17:8
_.mixin/</_.prototype[name]@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:69:526
eval_contexts@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:408:136
pyeval@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:427:8
eval@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:606:398
_getContext@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:1449:16
_readMissingFields@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:1489:85
_readUngroupedList/<@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:1500:13
then/</</<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:802:681
fire@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:796:299
add@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:797:471
then/</<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:802:649
each@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:625:767
then/<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:802:560
Deferred@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:803:194
then@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:802:525
_readUngroupedList@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:1498:12
_fetchX2Manys/<@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:1428:1195
_.forEach@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:12:566
_fetchX2Manys@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:1428:345
_fetchRecord/<@http://127.0.0.1:8269/web/content/4862-3f965a9/web.assets_backend.js:1414:172
then/</</<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:802:681
fire@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:796:299
fireWith@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:801:202
then/</</<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:802:874
fire@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:796:299
fireWith@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:801:202
then/</</<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:802:874
fire@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:796:299
fireWith@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:801:202
Deferred/</deferred[tuple[0]]@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:803:56
rpc/<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:3589:448
then/</</<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:802:681
fire@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:796:299
fireWith@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:801:202
Deferred/</deferred[tuple[0]]@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:803:56
fire@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:796:299
fireWith@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:801:202
Deferred/</deferred[tuple[0]]@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:803:56
genericJsonRpc/<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:3558:10
then/</</<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:802:681
fire@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:796:299
fireWith@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:801:202
then/</</<@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:802:874
fire@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:796:299
fireWith@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:801:202
done@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:1192:95
callback@http://127.0.0.1:8269/web/content/4861-75f87c7/web.assets_common.js:1212:23
```

It seems issue is on this line:

https://github.com/OCA/partner-contact/blob/735e86f2d1b482f437070c4445dd6f2628448f06/partner_default_sale_discount/models/sale_order.py#L36

error raised on empty `context`, because of `{`  replaced by `{'default_discount': default_sale_discount, `. Latest `,` generate it.

**Comment by pedrobaeza** [comment]
In the previous line, if `context` not found, the fallback value is `{}`, so it's strange that the problem comes from there.

**Comment by GSLabIt** [comment]
@pedrobaeza yap, u r right but at the end string is `{'default_discount': default_sale_discount, }` and raised the error; at least that's what i found. Any hint appreciated.

**Comment by pedrobaeza** [comment]
And you say that the final comma is provoking the problem?

**Comment by GSLabIt** [comment]
> And you say that the final comma is provoking the problem?

yes, that's what I've seen; maybe  string need to be converted to dict?

**Comment by pedrobaeza** [comment]
I don't think so, but please remove the comma and try again to see.

**Comment by GSLabIt** [comment]
@pedrobaeza i think i found the real issue. It's an incompatibility related to `sale_commission`:

https://github.com/OCA/commission/blob/c38d3c70d1d36c1318024b5e8e31d01cf595ce78/sale_commission/models/sale_order.py#L60. 

Fix at #1062
